### PR TITLE
Toolchain: Update LLVM to 15.0.3

### PR DESCRIPTION
**Pull request by timschumi**
LLVM 15 made some changes to how errors about C99-removed features (functions without declarations, optional `int` return type for the main function, etc.) are handled, making them hard-fails by default. In turn, this broke a significant portion of our ports, be it either because they didn't care to include all necessary headers or because we are missing some header dependency connection.

Either way, those hard-fails seem to have been relaxed somewhere between the initial release of LLVM 15 and LLVM 15.0.3, so let's just upgrade to that instead of fixing the affected ports manually (which would imply a lot of patch noise, and we probably wouldn't even find all the errors, as they might be buried somewhere in a configure script).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ endif()
 
 # Check for toolchain mismatch, user might need to rebuild toolchain
 set(GCC_VERSION "12.2.0")
-set(LLVM_VERSION "15.0.0")
+set(LLVM_VERSION "15.0.3")
```
Comment (timschumi) [inline CMakeLists.txt:21]: Do we maybe want to turn this into a `15.*` check (at least for this cycle), considering that not everyone cares about ports as much as I do?

Comment (ADKaster) [reply CMakeLists.txt:21]: I think one of the things this used to be used for was an actual path to some libs or headers, though I think we've removed all the fun -nodefaultlibs from the repo

Comment (timschumi) [reply CMakeLists.txt:21]: It doesn't seem like it is currently used in any logic apart from checking that the toolchain version matches what is expected.

**Review by BertalanD — APPROVED** [review]
Thanks!

**Pull request merged**